src/main/java/seedu/guest/logic/parser/AddCommandParser.java

```diff
@@ -46,29 +46,20 @@ public AddCommand parse(String args) throws ParseException {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+        Room room = ParserUtil.parseRoom(argMultimap.getValue(PREFIX_ROOM).get());
+        DateRange dateRange = ParserUtil.parseDateRange(argMultimap.getValue(PREFIX_DATE_RANGE).get());
+        NumberOfGuests numberOfGuests = ParserUtil
+                .parseNumberOfGuests(argMultimap.getValue(PREFIX_NUMBER_OF_GUESTS).get());
+        IsRoomClean isRoomClean = new IsRoomClean(DEFAULT_IS_ROOM_CLEAN);
+        Bill bill = new Bill();
         if (arePrefixesPresent(argMultimap, PREFIX_REQUEST)) {
-            Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-            Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-            Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
-            Room room = ParserUtil.parseRoom(argMultimap.getValue(PREFIX_ROOM).get());
-            DateRange dateRange = ParserUtil.parseDateRange(argMultimap.getValue(PREFIX_DATE_RANGE).get());
-            NumberOfGuests numberOfGuests = ParserUtil
-                    .parseNumberOfGuests(argMultimap.getValue(PREFIX_NUMBER_OF_GUESTS).get());
-            IsRoomClean isRoomClean = new IsRoomClean(DEFAULT_IS_ROOM_CLEAN);
-            Bill bill = new Bill();
             Request request = ParserUtil.parseRequest(argMultimap.getValue(PREFIX_REQUEST).get());
             Guest guest = new Guest(name, phone, email, room, dateRange, numberOfGuests, isRoomClean, bill, request);
```

Comment:
The lines "Guest guest = new Guest(name, phone, email, room, dateRange, numberOfGuests, isRoomClean, bill, request);" and " return new AddCommand(guest);" are duplicated. Should we remove it from the if-else block?

Reply:
cannot, because it will cause error

Reply:
Perhaps you need to place the declaration of request outside of the if else block for it to work?